swagger.yaml

```diff
@@ -10,33 +10,26 @@ basePath: "/v1"
 schemes:
   - http
 tags:
-  - name: "hello"
+  - name: "session"
 paths:
-  /hello:
-    get:
+  /session:
+    post:
       tags:
-        - hello
-      summary: Example hello world endpoint
-      description: Returns an example payload.
+        - session
+      summary: Create a session endpoint
+      description: Creates a session and logs request back
       produces:
         - application/json
       responses:
-        200:
-          description: OK
+        201:
+          description: Created
           schema:
-            $ref: "#/definitions/HelloResponse"
-        500:
-          $ref: '#/responses/InternalError'
+            $ref: "#/definitions/SessionResponse"
+        400:
+          description: Bad Request
+          schema:
+            $ref: "#/responses/BadRequest"
```

Comment:
Missing definition of BadRequest in responses object - you could remove this ass you have added a description directly on the status code 400

The reason for moving it to responses object and then referencing it is so it can be re-used by other parts of the spec, e.g. used on another endpoint

Reply:
You also need to add the type for the request body.